File: Codigos por competidor/Andres Espitia/DFS.java
import java.util.Iterator;
import java.util.LinkedList;
import java.util.Scanner;

public class DFS {
	static class Grafo {
		int nodos;
		LinkedList<Integer> adj[];
		
		Grafo(int v){
			this.nodos = v;
			adj = new LinkedList[this.nodos];
			for (int i = 0; i < adj.length; i++) {
				adj[i] = new LinkedList<Integer>();
			}
		}
		
		void agregarCamino(int desde, int hasta) {
			adj[desde].add(hasta);
		}
		
		void DFS(int origen) {
			boolean visitado[] = new boolean[nodos];
			DFSUtil(origen, visitado);
		}
		
		void DFSUtil(int actual, boolean visitado[]) {
			visitado[actual] = true;
			System.out.print(actual+" -> ");
			Iterator<Integer> i = adj[actual].listIterator();
			while(i.hasNext()) {
				int nuevoactual = i.next();
				if(!visitado[nuevoactual]) {
					DFSUtil(nuevoactual, visitado);
				}
			}
		}
	}
	
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int nodos = sc.nextInt();
		int caminos = sc.nextInt();
		
		Grafo g = new Grafo(nodos);
		for (int i = 0; i < caminos; i++) {
			int desde = sc.nextInt(), hasta = sc.nextInt();
			g.agregarCamino(desde, hasta);
		}
		g.DFS(sc.nextInt());
	}
	
	
}

File: Codigos por competidor/Diego Rodriguez/Algoritmos/DFS.kt
import java.util.*

object DFS {
    var sc = Scanner(System.`in`)
    @JvmStatic
    fun main(args: Array<String>) {
        val g = Graph(4)
        g.addEdge(0, 1)
        g.addEdge(0, 2)
        g.addEdge(1, 2)
        g.addEdge(2, 0)
        g.addEdge(2, 3)
        g.addEdge(3, 3)
        val inicial = 1
        println("Siguiendo la DFS desde el nodo $inicial")
        g.DFS(inicial)
    }

    internal class Graph(  //nodos
        var V: Int
    ) {
        var adj: Array<LinkedList<Int>>
        fun addEdge(v: Int, w: Int) {
            adj[v].add(w)
        }

        fun DFSUtil(v: Int, visited: BooleanArray) {
            visited[v] = true
            print("$v ")
            val i: Iterator<Int> = adj[v].listIterator()
            while (i.hasNext()) {
                val n = i.next()
                if (!visited[n]) {
                    DFSUtil(n, visited)
                }
            }
        }

        fun DFS(v: Int) {
            val visited = BooleanArray(V)
            DFSUtil(v, visited)
        }

        // Constructor
        init {
            adj = Array(V){i->LinkedList<Int>()
            }
        }
    }
}